data/src/main/java/com/example/data/implementationRepo/LastChosenCitiesRepoImpl.kt

```diff
@@ -0,0 +1,45 @@
+package com.example.data.implementationRepo
+
+import android.util.Log
+import com.example.data.db.dao.CityDao
+import com.example.data.db.dao.TableSizeDeleteCityDao
+import com.example.data.mappers.LastChosenCitiesEntityMapper
+import com.example.data.utils.MAX_TABLE_SIZE
+import com.example.domain.models.CityModel
+import com.example.domain.repositories.LastChosenCitiesRepo
+import io.reactivex.Single
+import io.reactivex.disposables.Disposable
+import io.reactivex.schedulers.Schedulers
+
+class LastChosenCitiesRepoImpl(
+    private val cityDao: CityDao,
+    private val mapper: LastChosenCitiesEntityMapper,
+    private val tableSizeDeleteCityDao: TableSizeDeleteCityDao
+) : LastChosenCitiesRepo {
+
+    override fun getLastChosenCities(): Single<List<CityModel>> {//This method gets list of last ten chosen cities from db and maps the list to list of CityModels
+        return cityDao.getAllLastChosenCities()
+            .subscribeOn(Schedulers.io())
+            .map { mapper.fromEntityToCityModelList(it) }
+    }
+
+    override fun insertCityToLastChosenCitiesEntity(cityId: Int, cityList: List<CityModel>?): Disposable {
+
+        return Single.just(cityId)
+            .subscribeOn(Schedulers.io())
+            .subscribe(
+                { city_id ->
+                    val tableSize = tableSizeDeleteCityDao.getTableSize()
```

Comment:
This code is still not safe. To create a transaction here you can use database.runInTransaction. And you don't need TableSizeDeleteCityDao, probably.

And use fromCallable {  } for creating Single here.  
Method insertCityToLastChosenCitiesEntity should return Single (if you want to return something) or Completable (if you want just know that operation is complete), but not disposable. Subscribing should happen outside this method.

Reply:
Fixed